onchain/src/username_store.cairo

```diff
@@ -0,0 +1,89 @@
+use starknet::ContractAddress;
+
+pub mod UserNameClaimErrors {
+    pub const USERNAME_CLAIMED: felt252 = 'username_claimed';
+    pub const USERNAME_CANNOT_BE_TRANSFER : felt252 = 'username_cannot_be_transferred';
+}
+
+#[starknet::interface]
+pub trait IUsernameStore<TContractState> {
+    fn claim_username(ref self: TContractState, key: felt252);
+    fn transfer_username(ref self: TContractState, key: felt252, new_Address: ContractAddress);
+}
+
+#[starknet::contract]
+pub mod UsernameStore {
+    use starknet::{get_caller_address, ContractAddress};
+    
+    use super::UserNameClaimErrors;
+
+    #[storage]
+    struct Storage {
+        username: LegacyMap::<felt252, ContractAddress>
```

Comment:
Please name this `usernames` for consistency with other modules.

Reply:
yaah, best we make it usernames